src/planner/plannodes/aggregate_plan_node.cpp

```diff
@@ -136,6 +136,20 @@ std::vector<std::unique_ptr<parser::AbstractExpression>> AggregatePlanNode::From
   return exprs;
 }
 
+bool AggregatePlanNode::RequiresCleanup() const {
+  return std::any_of(aggregate_terms_.begin(), aggregate_terms_.end(),
+                     [](auto agg_term) { return agg_term->RequiresCleanup(); });
+}
+std::vector<size_t> AggregatePlanNode::GetMemoryAllocatingAggregatorIndexes() const {
```

Comment:
Based on my understanding of this functionality, we are only ever going to call this member when tearing down the state for a particular pipeline, so it will only ever be called once by definition. If this is the case, it obviously makes sense to construct the vector when this is called, but would there ever be any situation where this might be called more than once for the same instance, making it worthwhile to build the vector once and cache it for later? Only alternative I can think of would be to make this return some lazy iterable, but that would almost certainly be overkill.

Reply:
This method will should only ever get called once per instance. Though now that I'm looking at it it gets called multiple times, which I'm going to fix so it only gets called once. This whole method might be overkill tbh because we can just have the translators iterate through the expressions themselves and see which need cleanup. Let me know your thoughts if you think this method is necessary.

Reply:
I am partial to overkill so my vote is for keeping it! Anything that contributes to expressing what the code is actually doing is a net win in my book, and this member does that for me.